From 2676119ddbb9387ac02a81063fae264b0b6ada86 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Sat, 22 May 2021 19:18:34 +0300
Subject: [PATCH 110/127] arm64: dts: msm8998-oneplus-cheeseburger: Configure
 RMI4 F1A buttons

This allows the capacitive buttons below the touchscreen to be used on
e.g. Android normally :)
---
 arch/arm64/boot/dts/qcom/msm8998-oneplus-cheeseburger.dts | 8 ++++++++
 arch/arm64/boot/dts/qcom/msm8998-oneplus-common.dtsi      | 2 +-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/msm8998-oneplus-cheeseburger.dts b/arch/arm64/boot/dts/qcom/msm8998-oneplus-cheeseburger.dts
index ef2a88a64d32..f36f4daa233c 100644
--- a/arch/arm64/boot/dts/qcom/msm8998-oneplus-cheeseburger.dts
+++ b/arch/arm64/boot/dts/qcom/msm8998-oneplus-cheeseburger.dts
@@ -41,3 +41,11 @@ pinconf {
 		};
 	};
 };
+
+/* Capacitive keypad buttons */
+&rmi4_i2c {
+	rmi4-f1a@1a {
+		reg = <0x1a>;
+		syna,codes = <KEY_BACK KEY_APPSELECT>;
+	};
+};
diff --git a/arch/arm64/boot/dts/qcom/msm8998-oneplus-common.dtsi b/arch/arm64/boot/dts/qcom/msm8998-oneplus-common.dtsi
index c028632b30b5..58862c04fb1b 100644
--- a/arch/arm64/boot/dts/qcom/msm8998-oneplus-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8998-oneplus-common.dtsi
@@ -169,7 +169,7 @@ &wlan_msa_mem {
 &blsp1_i2c5 {
 	status = "okay";
 
-	touchscreen@20 {
+	rmi4_i2c: touchscreen@20 {
 		compatible = "syna,rmi4-i2c";
 		reg = <0x20>;
 		#address-cells = <1>;
-- 
2.38.1

